File: exercism/cpp/prime-factors/prime_factors.cpp
#include "prime_factors.h"

std::vector<int> prime_factors::of(int n) {
  std::vector<int> factors;
  int k = 2;

  while (n >= k) {
    if (n % k == 0) {
      factors.push_back(k);
      n /= k;
    } else {
      k += 1;
    }
  }

  return factors;
}

File: exercism/go/prime-factors/prime_factors.go
package prime

// Factors returns the prime factors of `n` in ascending order
func Factors(n int64) []int64 {
	var k int64 = 2
	factors := make([]int64, 0)
	for n > 1 {
		if n%k == 0 {
			factors = append(factors, k)
			n = n / k
		} else {
			k++
		}
	}
	return factors
}
